ietf/meeting/management/commands/generate_schedule.py

```diff
@@ -43,32 +63,60 @@ def add_arguments(self, parser):
         parser.add_argument('-r', '--max-runs', type=int, dest='max_cycles',
                             default=OPTIMISER_MAX_CYCLES,
                             help='maximum optimiser runs')
+        parser.add_argument('-b', '--base-schedule',
+                            type=ScheduleId.from_str,
+                            dest='base_id',
+                            default=None,
+                            help=(
+                                'Base schedule for generated schedule, specified as "[owner/]name"'
+                                ' (default is no base schedule; owner not required if name is unique)'
+                            ))
 
-    def handle(self, meeting, name, max_cycles, verbosity, *args, **kwargs):
-        ScheduleHandler(self.stdout, meeting, name, max_cycles, verbosity).run()
+    def handle(self, meeting, name, max_cycles, verbosity, base_id, *args, **kwargs):
+        ScheduleHandler(self.stdout, meeting, name, max_cycles, verbosity, base_id).run()
 
 
 class ScheduleHandler(object):
-    def __init__(self, stdout, meeting_number, name=None, max_cycles=OPTIMISER_MAX_CYCLES, verbosity=1):
+    def __init__(self, stdout, meeting_number, name=None, max_cycles=OPTIMISER_MAX_CYCLES,
+                 verbosity=1, base_id=None):
         self.stdout = stdout
         self.verbosity = verbosity
         self.name = name
         self.max_cycles = max_cycles
+
         if meeting_number:
             try:
                 self.meeting = models.Meeting.objects.get(type="ietf", number=meeting_number)
             except models.Meeting.DoesNotExist:
                 raise CommandError('Unknown meeting number {}'.format(meeting_number))
         else:
             self.meeting = models.Meeting.get_current_meeting()
+
+        if base_id is None:
+            self.base_schedule = None
```

Comment:
It would for that purpose, but there are a few places where it's tested against `None` to decide whether to include a base schedule (e.g., when loading or saving). I think the out-of-band indicator that there is no base schedule is safer than having to remember to check whether the schedule exists when it matters.